dace/libraries/blas/nodes/ger.py

```diff
@@ -0,0 +1,403 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+from dace.symbolic import symstr
+from dace.properties import Property, SymbolicProperty
+from dace.transformation.transformation import ExpandTransformation
+from dace.frontend.common import op_repository as oprepo
+from dace.sdfg.nodes import LibraryNode
+from dace.libraries.blas.nodes.matmul import _get_matmul_operands
+import dace.library as library
+from dace.sdfg import SDFG, SDFGState, nodes
+from dace import data as dt, memlet as mm, subsets as sbs
+import dace
+import copy
+import numpy as np
+
+import dace.library
+import dace.properties
+import dace.sdfg.nodes
+
+from dace import dtypes
+from dace.memlet import Memlet
+
+
+@library.expansion
+class ExpandGerPure(ExpandTransformation):
+
+    environments = []
+
+    @staticmethod
+    def expansion(node, parent_state, parent_sdfg, **kwargs):
+        node.validate(sdfg, state)
+        inputs = ('_A', '_x', '_y')
+        outputs = ('_res', )
+        in_edges = [
+            next(parent_state.in_edges_by_connector(node, conn))
+            for conn in inputs
+        ]
+        out_edges = [
+            next(parent_state.out_edges_by_connector(node, conn))
+            for conn in outputs
+        ]
+        arrays = {}
+        arrays.update({
+            inp: parent_sdfg.arrays[e.data.data]
+            for inp, e in zip(inputs, in_edges)
+        })
+        arrays.update({
+            out: parent_sdfg.arrays[e.data.data]
+            for out, e in zip(outputs, out_edges)
+        })
+
+        # TODO: Support memlet subsets
+        if any(e.data.subset != sbs.Range.from_array(arrays[a])
+               for a, e in zip(inputs, in_edges)):
+            raise NotImplementedError
+        if any(e.data.subset != sbs.Range.from_array(arrays[a])
+               for a, e in zip(outputs, out_edges)):
+            raise NotImplementedError
+
+        sdfg = dace.SDFG(f'{node.label}_sdfg')
+        sdfg.add_symbol('M', int)
+        sdfg.add_symbol('N', int)
+        sdfg.add_symbol('alpha', arrays['_A'].dtype)
```

Comment:
This is a different approach used by Tal, I don't want to change it now because I don't actually know what the correct approach is. It's complicated because of many different ways the sizes can be provided.